docs/utilities/batch.md

```diff
@@ -522,14 +522,20 @@ You might want to bring custom logic to the existing `BatchProcessor` to slightl
 
 For these scenarios, you can subclass `BatchProcessor` and quickly override `success_handler` and `failure_handler` methods:
 
-* **`success_handler()`** – Keeps track of successful batch records
-* **`failure_handler()`** – Keeps track of failed batch records
+* **`success_handler()`** is called for each successfully processed record
+* **`failure_handler()`** is called for each failed record
+
+???+ warning
+    These functions have a common `record` argument. For backward compatibility reasons, their type is not the same:
+
+    - `success_handler`: `record` type is `dict[str, Any]`, the raw record data.
+    - `failure_handler`: `record` type can be an Event Source Data Class or your [Pydantic model](#pydantic-integration). During Pydantic validation errors, we fall back and serialize `record` to Event Source Data Class to not break the processing pipeline.
 
 ???+ example
-	Let's suppose you'd like to add a metric named `BatchRecordFailures` for each batch record that failed processing
+	Let's suppose you'd like to add metrics to track successes and failures of your batch records.
```

Comment:
```suggestion
Let's suppose you'd like to add metrics to track successes and failures of your batch records.
```

Reply:
Just noticed before I merged that we have two callouts, this can be noisy for some readers. The callout you just added is more important - I'll suggest moving to Note too so it's less of a danger-feel

<img width="1200" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/3340292/70c643b3-0352-47c4-afec-7a36d287d524">